spec/src/main/asciidoc/core/spi.asciidoc

```diff
@@ -697,6 +697,7 @@ The method `BeanManager.getInterceptionFactory()` returns an `InterceptionFactor
 <T> InterceptionFactory<T> createInterceptionFactory(CreationalContext<T> ctx, Class<T> clazz);
 ----
 
+If type parameter `T` is not a class non-portable behavior results.
```

Comment:
That would be truly weird use case. I mean can you even make a producer for enum so that you would want to intercept it?